array.h

```diff
@@ -1527,6 +1527,10 @@ NDARRAY_HOST_DEVICE auto make_compact(const Shape& s) {
   return static_compact;
 }
 
+/** A `shape` where all extents and strides are constant. */
+template <index_t... Extents>
+using fixed_dense_shape = decltype(make_shape_from_tuple(internal::make_compact_dims<1>(dim<0, Extents>()...)));
+
```

Comment:
I guess it's not trivial (or necessary) to make a version that lets you specify mins as well.

Reply:
I thought about it, but I can't think of a good API to support it, and I also don't think it's necessary.